comps/embeddings/predictionguard/embedding_pg.py

```diff
@@ -0,0 +1,37 @@
+import os
+import time
+
+from predictionguard import PredictionGuard
+from comps import EmbedDoc512, ServiceType, TextDoc, opea_microservices, register_microservice, statistics_dict, register_statistics
+
+# Initialize Prediction Guard client
+client = PredictionGuard()
+
+@register_microservice(
+    name="opea_service@embedding_predictionguard",
+    service_type=ServiceType.EMBEDDING,
+    endpoint="/v1/embeddings",
+    host="0.0.0.0",
+    port=6000,
+    input_datatype=TextDoc,
+    output_datatype=EmbedDoc512,
+)
+
+@register_statistics(names=["opea_service@embedding_predictionguard"])
+def embedding(input: TextDoc) -> EmbedDoc512:
```

Comment:
Do the other service options have the ability to "batch" multiple text inputs into a single request? Or are they setup like this? Technically we could have multiple in a single request.

Reply:
all the others are setup in similar fashion.